core/vocabs/loader/loader.factor

```diff
@@ -56,7 +56,8 @@ PRIVATE>
     ] cache ;
 
 : vocab-exists? ( name -- ? )
-    dup lookup-vocab [ ] [ find-vocab-root ] ?if ;
+    [ dup lookup-vocab [ ] [ find-vocab-root ] ?if ]
+    [ dup bad-vocab-name? [ 2drop f ] [ rethrow ] if ] recover ;
```

Comment:
I'd prefer not to use ``recover`` here.  It looks like ``find-vocab-root`` calls ``vocab-name``, which throws.  Perhaps we make a ``?vocab-name`` which returns ``name/f`` and then ``find-vocab-root`` can skip the finding work if not found?

Other suggestions?